components/dashboard/src/user-settings/Preferences.tsx

```diff
@@ -29,6 +31,34 @@ export default function Preferences() {
         }
     };
 
+    const [disabledClosedTimeout, setDisabledClosedTimeout] = useState<boolean>(
+        user?.additionalData?.disabledClosedTimeout ?? false,
+    );
+    const actuallySetDisabledClosedTimeout = async (value: boolean) => {
+        try {
+            await getGitpodService().server.updateWorkspaceTimeoutSetting({ disabledClosedTimeout: value });
+            setDisabledClosedTimeout(value);
+        } catch (e) {
+            alert("Cannot set custom workspace timeout: " + e.message);
+        }
+    };
+
+    const [workspaceTimeout, setWorkspaceTimeout] = useState<string>(user?.additionalData?.workspaceTimeout ?? "");
+    const actuallySetWorkspaceTimeout = async (value: string) => {
+        try {
+            await getGitpodService().server.updateWorkspaceTimeoutSetting({ workspaceTimeout: value });
+        } catch (e) {
+            alert("Cannot set custom workspace timeout: " + e.message);
```

Comment:
Is there a better way to communicate this rather than showing the `alert` box? It's rather un-friendly  and has the potential of spamming the user if the error happens to trigger on a loop. @gtsiolis what's a good pattern to communicate an API error here?

Reply:
@easyCZ Yes:
1. We could use an in-page alert as we do for profile updates or paused prebuilds, see screenshots below.
2. Alternatively, using a form validation error like we do with the onboarding could be nice. Cc @selfcontained because https://github.com/gitpod-io/gitpod/pull/16501.
3. Long-term, I'd like to see this moving away from this custom value and rely on just a number, making this a number type of field. It's quite confusing and hard to understand what's going on here, whereas if we just used minutes as we do with prebuilds it could simplify things, including what copy we use for labels and input help text. Cc @loujaybee 

However, I'd not block merging this as is for now, even if it's just a browser alert.

Thanks for the ping, @easyCZ! 🏓 

| Profile update | Paused prebuilds |
|-|-|
| <img width="609" alt="Screenshot 2023-02-28 at 18 02 07" src="https://user-images.githubusercontent.com/120486/222213736-0b225974-5663-4453-b1da-ef3001d0e790.png"> | <img width="1127" alt="Screenshot 2023-03-01 at 19 15 58" src="https://user-images.githubusercontent.com/120486/222213635-e357e8c8-1203-47e5-9ea7-4599b8ca6cd3.png"> |

Reply:
I also don't think we should really ever be using a native browser `alert()`, for reasons @easyCZ described, and it also communicates a very unpolished feel. Showing an inline `<Alert>` message in the UI next to the place the button or form lives seems like a decent approach. If we had toast notifications, that would be nice as well, but we don't atm.